semanticdb/metadiff/src/main/scala/scala/meta/internal/metadiff/diff/package.scala

```diff
@@ -0,0 +1,84 @@
+package scala.meta.internal.metadiff
+
+import difflib.DiffUtils
+import java.util.regex.Pattern
+import scala.collection.JavaConverters._
+
+package object diff {
+
+  val EOL = System.lineSeparator()
+
+  type Diff = Option[String]
+
+  implicit class DiffOps(diff: Diff) {
+
+    def name(name: String): Diff =
+      diff.map { d =>
+        val headerBar = "=" * name.length
+        s"$name$EOL" +
+          s"$headerBar$EOL" +
+          d
+      }
+
+    def concat(that: Diff): Diff = diffConcat(diff, that)
+
+    def extractLines: Option[Seq[String]] = diff.map { text =>
+      text.lines
+        .filterNot(_.startsWith("@@"))
+        .map { line =>
+          line.substring(1)
+        }
+        .toSeq
+    }
+
+  }
+
+  def diffLines(
+      linesFrom: TraversableOnce[String],
+      linesTo: TraversableOnce[String],
+      contextSize: Int = 1000000): Diff = {
+    val origLines = linesFrom.toSeq.asJava
+    val patch = DiffUtils.diff(origLines, linesTo.toSeq.asJava)
+    if (patch.getDeltas.isEmpty) None
+    else {
+      val diffStr = DiffUtils
+        .generateUnifiedDiff("", "", origLines, patch, contextSize)
+        .asScala
+        .drop(3)
+        .mkString(EOL)
+      Some(diffStr + EOL)
+    }
+  }
+
+  def diffObj[T](objFrom: T, objTo: T): Diff =
```

Comment:
How about calling this `diffStrings` instead? Genericity of the signature may give off an impression that the method is typeclass-based.